piet/src/gradient.rs

```diff
@@ -50,3 +68,168 @@ pub struct GradientStop {
     /// The color at that stop.
     pub color: Color,
 }
+
+/// A flexible, ergonomic way to describe gradient stops.
+pub trait GradientStops {
+    fn to_vec(self) -> Vec<GradientStop>;
+}
+
+/// A description of a linear gradient in the unit rect, which can be resolved
+/// to a concrete gradient.
+///
+/// The start and end points in the gradient are given in [`UnitPoint`] coordinates,
+/// which are then resolved to image-space coordinates for any given concrete `Rect`.
+///
+/// When the fixed coordinates are known, use [`FixedGradient`] instead.
+///
+/// [`UnitPoint`]: struct.UnitPoint.html
+/// [`FixedGradient`]: struct.FixedGradient.html
+pub struct LinearGradient {
+    start: UnitPoint,
+    end: UnitPoint,
+    stops: Vec<GradientStop>,
+}
+
+/// A representation of a point relative to a rectangle.
+pub struct UnitPoint {
+    u: f64,
```

Comment:
I'm curious why this is `u, v` and not `x, y`, assume there's some mathy precedent?

Reply:
Both would work, but u, v is common for texture coords normalized to the unit square.